### DISCUSSION TITLE
File Encryption

### DISCUSSION THREAD

**Discussion by Annabach0** [stashapp/stash#4362] [Q&A] [answered]
Can the Images and Videos be encrypted, so that they can not be seen as such in the file system?

**Comment by dingdongsolong4** [comment, marked as answer]
This would be possible, but it would in my opinion make more sense to do encryption with an external program, outside of Stash. The program would mount a virtual filesystem somewhere, and all reads and writes would be transparently decrypted/encrypted before actually being stored on your real filesystem. You'd then simply set your Stash library path to the virtual filesystem.

There are many programs that can do this. [Rclone](https://rclone.org/) is one such program, and while it is mostly intended for working with remote files, it can also be used to add encryption to local files as well.

#3858 is a similar request.

**Comment by joewood9364** [comment]
Would be nice if generated content/blobs/database was encrypted at rest while the media sources themselves are left untouched. I would be comfortable keeping the stash application in cloud storage if this was implemented.

**Reply by dingdongsolong4** [reply]
This should also easily be possible with something like Rclone.

But **please please** don't **ever** store your database in cloud storage - you will almost certainly corrupt it. SQLite does not work over a network, or on anything that isn't purely local storage.

Blobs should be fine to store anywhere, but they are accessed quite frequently so it would be much better for responsiveness if those were local as well. Generated content should also be fine to store anywhere.
